### DISCUSSION TITLE
Set custom color legend sort order when using stacked bars

### DISCUSSION THREAD

**Discussion by r00tnpm** [vega/vega-lite#9482] [Q&A] [answered]
hi team, new to this, may need some help with stacked bar chart, using the example below: 

- scale/domain sort legend, but not the color-bar itself 
- for example, in the legend, "sun" = 1, "snow" = 5, but the "snow" color-bar is next to "sun" on chart

my question is how do I sort the bars in the same order as legend? 

https://vega.github.io/vega-lite/examples/stacked_bar_weather.html

thanks!

**Comment by hydrosquall** [comment, marked as answer]
Hi @r00tnpm , you can sort the entries based on how you define the legend domain order ([example](https://vega.github.io/editor/#/url/vega-lite/N4IgJAzgxgFgpgWwIYgFwhgF0wBwqgegIDc4BzJAOjIEtMYBXAI0poHsDp5kTykBaADZ04JAGyUAVhDYA7EABoQAEySYUqUAwBOgtCrVJOcNZkFx+AdxP042ylAjEQAXyXJtAa31Mk2xSBwslBsyjSyZGigAB5RIJg0CHAAqrJ0+ghy9AEAZjRwgsr6qphwAZgAnjhl6GzaYbJIekoJZjUgALJZMAAEbDk9tj0VJv5uIBVxSGRk2nyl+iEMspjlVe0AjgxIK3RqNKSuSiGCdXF5BUXo1mrw-i3r+rJsCOFNAdBNNaDKL0jhaAA2iptDQAF5g8y5NiRJTaf7yJQQZ6WD7LEAAXThOzINWBAGIkHAoAB2OAADgC+NJNJJVIAjDkSSSmAAWKkATlZYhZRSU+LgLKQAFYAEyY8YybSrTQgNg4RZsZarcatKHoADqNjug0eLn1QA) )

<img width="1062" alt="image" src="https://github.com/user-attachments/assets/d61f4b7d-d61b-4b02-91ce-35f563f09784" />

In the more general case where you didn't set a custom scale, see @PBI-David 's answer. This works when you don't have a custom domain and range: https://stackoverflow.com/a/76158192/5129731
